cpp/src/parquet/statistics_test.cc

```diff
@@ -71,19 +71,25 @@ static FLBA FLBAFromString(const std::string& s) {
 TEST(Comparison, SignedByteArray) {
   auto comparator = MakeComparator<ByteArrayType>(Type::BYTE_ARRAY, SortOrder::SIGNED);
 
-  std::string s1 = "12345";
-  std::string s2 = "12345678";
-  ByteArray s1ba = ByteArrayFromString(s1);
-  ByteArray s2ba = ByteArrayFromString(s2);
-  ASSERT_TRUE(comparator->Compare(s1ba, s2ba));
+  std::vector<uint8_t> byte_values[] = {
+      {0x80, 0x80, 0, 0},       {/*0xFF,*/ 0x80, 0, 0},     {/*0xFF,*/ 0xFF, 0x01, 0},
+      {/*0xFF,0xFF,*/ 0x80, 0}, {/*0xFF,0xFF,0xFF,*/ 0x80}, {/*0xFF, 0xFF, 0xFF,*/ 0xFF},
+      {/*0, 0,*/ 0x01, 0x01},   {/*0,*/ 0x01, 0x01, 0},     {0x01, 0x01, 0, 0}};
```

Comment:
Can you also test pairs of equal values with differing lengths?
Perhaps even do exhaustive pairwise comparisons, e.g.:
```c++
struct Case {
  std::vector<uint8_t> bytes;
  int order;
};
std::vector<Case> cases[] = {
  {{0x80, 0x80, 0, 0}, 1},
  {{0xFF, 0x80, 0x80, 0, 0}, 1},
  {{0x80, 0, 0}, 2},
  {{0xFF, 0x80, 0, 0}, 2},
  {{0xFF, 0xFF, 0x80, 0, 0}, 2},
  {{0xFF, 0x01, 0}, 3},
  {{0xFF, 0xFF, 0x01, 0}, 3},
  // ...
};
for (const auto& a : cases) {
  for (const auto& b : cases) {
    ByteArray ba(static_cast<int>(a.bytes.size()), a.bytes.data()));
    ByteArray bb(static_cast<int>(b.bytes.size()), b.bytes.data()));
    if (a.order < b.order) {
      EXPECT_TRUE(comparator->Compare(ba, bb)) << ba << " " << bb;
    } else {
      EXPECT_FALSE(comparator->Compare(ba, bb)) << ba << " " << bb;
    }
  }
}
```

Reply:
did something similar to this.  I don't think ByteArray supports string insertion?